applications/services/desktop/views/desktop_view_lock_menu.c

```diff
@@ -4,14 +4,221 @@
+}
+
+FuriString* desktop_lock_menu_dummy_mode_get_text(
+    const DesktopLockMenuItem* item,
+    DesktopLockMenuViewModel* model,
+    bool is_selected) {
+    furi_assert(item);
+    furi_assert(model);
+    UNUSED(is_selected);
+
+    if(model->dummy_mode)
+        return furi_string_alloc_set("Default Mode");
+    else
+        return furi_string_alloc_set("Dummy Mode");
+}
+
+bool desktop_lock_menu_dummy_mode_on_input(
+    const DesktopLockMenuItem* item,
+    const InputEvent* event,
+    DesktopLockMenuView* view) {
+    furi_assert(item);
+    furi_assert(event);
+    furi_assert(view);
+
+    if(event->key == InputKeyOk) {
+        DesktopEvent event_type = DesktopLockMenuEventDummyModeOff;
+        with_view_model(
+            view->view,
+            DesktopLockMenuViewModel * model,
+            {
+                if(!model->dummy_mode) event_type = DesktopLockMenuEventDummyModeOn;
+            },
+            false)
+
+            view->callback(event_type, view->context);
+
+        return true;
+    } else
+        return false;
+}
+
+FuriString* desktop_lock_menu_display_brightness_get_text(
+    const DesktopLockMenuItem* item,
+    DesktopLockMenuViewModel* model,
+    bool is_selected) {
+    furi_assert(item);
+    furi_assert(model);
+
+    const char* value_string = backlight_text[model->display_brightness_index];
+    FuriString* str = furi_string_alloc_set("Display ");
+
+    if(is_selected) {
+        if(model->display_brightness_index > 0) furi_string_cat(str, "< ");
+
+        furi_string_cat(str, value_string);
+
+        if(model->display_brightness_index < BACKLIGHT_COUNT - 1) furi_string_cat(str, " >");
+    } else {
+        furi_string_cat(str, value_string);
+    }
+
+    return str;
+}
+
+bool desktop_lock_menu_display_brightness_on_input(
+    const DesktopLockMenuItem* item,
+    const InputEvent* event,
+    DesktopLockMenuView* view) {
+    furi_assert(item);
+    furi_assert(event);
+    furi_assert(view);
+
+    if(event->key == InputKeyLeft || event->key == InputKeyRight) {
+        bool update = false;
+        uint8_t index = 0;
+        with_view_model(
+            view->view,
+            DesktopLockMenuViewModel * model,
+            {
+                if(event->key == InputKeyLeft &&
+                   (int8_t)model->display_brightness_index - 1 >= 0) {
+                    model->display_brightness_index--;
+                    index = model->display_brightness_index;
+                    update = true;
+                } else if(
+                    event->key == InputKeyRight &&
+                    model->display_brightness_index + 1 < BACKLIGHT_COUNT) {
+                    model->display_brightness_index++;
+                    index = model->display_brightness_index;
+                    update = true;
+                }
+
+                if(update) {
+                    model->notification_settings_changed = true;
+                }
+            },
+            update)
```

Comment:
missing `;`

Reply:
Just out of curiosity, does the semicolon actually has.a use? As far as I, as a .NET none macro guy, it's just being append after the macro block and doesn't have a actual use.